keps/sig-auth/267-kubelet-server-certificate-bootstrap-rotation/README.md

```diff
@@ -0,0 +1,438 @@
+
+### Implementation
+
+[certificate](https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/certificate) contains the implementation of this feature.
+
+This features creates the object `NewKubeletServerCertificateManager`.
+
+### Metrics
+
+Three metrics were added for this feature.
+
+- "server_expiration_renew_errors"
+- "certificate_manager_server_rotation_seconds"
+- "certificate_manager_server_ttl_seconds"
+
+```golang
+compbasemetrics.NewCounter(
+  &compbasemetrics.CounterOpts{
+   Subsystem:      metrics.KubeletSubsystem,
+   Name:           "server_expiration_renew_errors",
+   Help:           "Counter of certificate renewal errors.",
+   StabilityLevel: compbasemetrics.ALPHA,
+  },
+ )
+
+compbasemetrics.NewHistogram(
+  &compbasemetrics.HistogramOpts{
+   Subsystem: metrics.KubeletSubsystem,
+   Name:      "certificate_manager_server_rotation_seconds",
+   Help:      "Histogram of the number of seconds the previous certificate lived before being rotated.",
+   Buckets: []float64{
+    60,        // 1  minute
+    3600,      // 1  hour
+    14400,     // 4  hours
+    86400,     // 1  day
+    604800,    // 1  week
+    2592000,   // 1  month
+    7776000,   // 3  months
+    15552000,  // 6  months
+    31104000,  // 1  year
+    124416000, // 4  years
+   },
+   StabilityLevel: compbasemetrics.ALPHA,
+  },
+  	legacyregistry.RawMustRegister(compbasemetrics.NewGaugeFunc(
+		&compbasemetrics.GaugeOpts{
+			Subsystem: metrics.KubeletSubsystem,
+			Name:      "certificate_manager_server_ttl_seconds",
+			Help: "Gauge of the shortest TTL (time-to-live) of " +
+				"the Kubelet's serving certificate. The value is in seconds " +
+				"until certificate expiry (negative if already expired). If " +
+				"serving certificate is invalid or unused, the value will " +
+				"be +INF.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		func() float64 {
+			if c := m.Current(); c != nil && c.Leaf != nil {
+				return math.Trunc(time.Until(c.Leaf.NotAfter).Seconds())
+			}
+			return math.Inf(1)
+		},
+	))
+```
+
+These metrics are all in alpha stage so it is recommended to promote these metrics to GA.
+
+These metrics should be promoted to [beta](https://github.com/kubernetes/kubernetes/pull/122834
+) before GA.
+
+### Test Plan
+
+Configuration testing is already done.
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+None
+
+##### Unit tests
+
+There is a unit test coverage for the certificate manager.
+
+- `pkg/kubelet/certificate`: `2024-01-17` - `52.1`
+
+##### Integration tests
+
+There are tests in `pkg/kubelet/certificate/bootstrap`.
+
+##### e2e tests
+
+Discussing this feature in sig-auth it was suggested that integration tests should be sufficient
+
+NA
```

Comment:
```suggestion
We won't do any end to end tests; this is because SIG Auth feel that integration tests provide sufficient
coverage.
```
?

I think an optional end to end test would be nice to have, even so.